fem/tmop_tools.cpp

```diff
@@ -336,19 +404,72 @@ double TMOPNewtonSolver::ComputeScalingFactor(const Vector &x,
 
 void TMOPNewtonSolver::ProcessNewState(const Vector &x) const
 {
-   if (discr_tc)
+   if (parallel)
    {
-      if (parallel)
-      {
 #ifdef MFEM_USE_MPI
-         const ParNonlinearForm *nlf =
-            dynamic_cast<const ParNonlinearForm *>(oper);
-         Vector x_loc(nlf->ParFESpace()->GetVSize());
-         nlf->ParFESpace()->GetProlongationMatrix()->Mult(x, x_loc);
-         discr_tc->UpdateTargetSpecification(x_loc);
+      const ParNonlinearForm *nlf =
+         dynamic_cast<const ParNonlinearForm *>(oper);
+      const Array<NonlinearFormIntegrator*> &integrators = *nlf->GetDNFI();
+      Vector x_loc(nlf->ParFESpace()->GetVSize());
+      nlf->ParFESpace()->GetProlongationMatrix()->Mult(x, x_loc);
+      const FiniteElementSpace *fesc = nlf->FESpace();
+      for (int i=0; i<integrators.Size(); i++)
+      {
+         const TMOP_Integrator *nlfi =
```

Comment:
If you don't declare it as `const` here, you won't have to do `const_cast` on the next line (the integrators are not `const` in the `Array` you're getting).